context.go

```diff
@@ -178,6 +178,9 @@ func (c *Context) SetHTTPRequest(req *http.Request) {
 		HTTPVersion: httpVersion,
 		Cookies:     req.Cookies(),
 	}
+	if c.requestBody.Form == nil && req.Form != nil {
+		c.requestBody.Form = req.Form
+	}
```

Comment:
Nice idea!

A few small issues (see SetHTTPRequestBody and BodyCapturer.setContext):
 - we should use req.PostForm instead of req.Form
 - we should be copying and truncating the values
 - we should honour c.captureBody (capturing body can be disabled through config)

My recommendation is that we record the net/http.Request on the context object, and then update SetHTTPRequestBody to pass it into BodyCapturer.setContext. Something like:

Here:
```go
    c.httpRequest = req // store for SetHTTPRequestBody
```

SetHTTPRequestBody
```go
    if bc.setContext(&c.requestBody, c.httpRequest) {
```

BodyCapturer.setContext:
```go
func (bc *BodyCapturer) setContext(out *model.RequestBody, req *http.Request) bool {
    // bc.request may have been copied, hence its PostForm will not be modified.
    // req is the request passed to the instrumented handler.
    if req.PostForm != nil {
        ...
```